File: Partition_Labels/Partition_Labels.go
package main

func main() {

	s := "ababcbacadefegdehijhklij"
	println(partitionLabels(s))
}

func partitionLabels(s string) []int {
	mk := make(map[string]int)
	for i, s := range s {
		mk[string(s)] = i
	}
	var m int
	var arr []int
	var res int
	for i, char := range s {
		m = maxNum(mk[string(char)], m)
		if i == m {
			arr = append(arr, m+1-res)
			res = m + 1
		}
	}
	return arr
}
func maxNum(i, j int) int {
	if i > j {
		return i
	}
	return j
}

File: Partition_Labels/Partition_Labels.py
s = "ababcbacadefegdehijhklij"

def partitionLabels(s):
    d={}
    for i in range(len(s)):
        d[s[i]] = i

    res = 0
    out = 0
    ls = []
    for i,char in enumerate(s):
        res = max(d[char],res)
        if i == res:
            ls.append(len(s[out:res+1]))
            out = res+1

    return ls



print(partitionLabels(s))